src/ElementFinderFactory.php

```diff
@@ -22,10 +22,13 @@
 class ElementFinderFactory implements ElementFinderFactoryInterface
 {
     public function __construct(
-        private readonly ?ExpressionTranslatorInterface $expressionTranslator = new XpathExpression(),
-        private readonly ?LinkConverterInterface $linkConverter = new RelativeToAbsoluteLinkConverter(),
-        private readonly ?EncodingConverterInterface $encodingConverter = new ToUtfConverter()
+        private ?ExpressionTranslatorInterface $expressionTranslator = null,
+        private ?LinkConverterInterface $linkConverter = null,
+        private ?EncodingConverterInterface $encodingConverter = null
     ) {
+        $this->expressionTranslator ??= new XpathExpression();
```

Comment:
this operator is available only from version 8.2 right?

Reply:
Null coalescing assignment operator is available from 7.4
https://wiki.php.net/rfc/null_coalesce_equal_operator